controllers/AdminController.php

```diff
@@ -301,16 +308,26 @@ public function actionInfo($id)
      */
     public function actionSwitch($id)
     {
-        $user = $this->findModel($id);
+        $old = Yii::$app->user;
+        $session = Yii::$app->session;
 
-        $old = \Yii::$app->user;
+        if($id == 'previous' && $session->has('previous_user')) {
+            $user = $this->findModel($session->get('previous_user'));
+            $session->remove('previous_user');
+        } else {
+            if (!Yii::$app->user->can('admin'))
```

Comment:
Please elaborate, i don´t really know what you mean with this one. 'Admin' is the default role in the whole yii2-user project. Making this configurable would be a bigger change IMHO and should be done outside of this PR.